cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -250,7 +250,8 @@ func (w *KubeWaiter) WaitForKubelet(healthzAddress string, healthzPort int32) er
 		healthzEndpoint = fmt.Sprintf("http://%s:%d/healthz", healthzAddress, healthzPort)
 	)
 
-	fmt.Printf("[kubelet-check] Waiting for a healthy kubelet. This can take up to %v\n", w.timeout)
+	fmt.Printf("[kubelet-check] Waiting for a healthy kubelet at %s. This can take up to %v\n",
+		healthzEndpoint, w.timeout)
```

Comment:
this is useful to have.

Reply:
Do we also want consider adding a check in here to ensure this healthz endpoint check is only done when the `HealthzPort` is set to a value other than ‘0’? If the `HealthzPort` is set to ‘0’ then healthz endpoint would be [disabled](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/config/v1beta1/types.go#L235) and this would cause this check to timeout. Apologies for any confusion on my part.

Reply:
makes sense to skip if the port is 0. i will update.